### ISSUE TITLE
Enigma: Missing passphrase prompt on message decryption

### ISSUE THREAD

**Issue by pravino** [roundcube/roundcubemail#5512]
Encryption Error- BOB to Alice

when sending mail to Alice through BOB enigma uses BOB's Key to Encrypt, so when login to Alice and when click on Email

it says BOB's Private key not available...
if i import BOBs Private key to Alice... then it decrypts by asking BOB's Passphrase

**Comment by CRtEurope** [comment]
Right, The same happens to me. All certificates do exist.
Branch master
PHP 7.0, partially 7.1
PGP 2.1

**Comment by alecpl** [comment]
You have both private keys in a keyring, right?

**Comment by pravino** [comment]
@alecpl yes i do..

**Comment by CRtEurope** [comment]
Yes, Alec

**Comment by alecpl** [comment]
@privano, you're also on git-master and Crypt_GPG 1.6.0b1? I can confirm the issue with this Crypt_GPG version.

On a side note, you have to aknowledge that the fact that a message is encrypted with both keys is correct. We do this so the sender is able to decrypt the sent message. So, if on the receiver end, you have both private keys (note private, not public here) then you'll be asked for a passphrase for any of them. We assume that if you have a private key in keyring you will know its passphrase (in gpg-2.1 you can't import private key without passphrase). I don't really know how to improve that.

Two issues have been described here:
1. When only one private key is in the keyring, there's "no private key" error. This I confirm.
2. When both private keys are in the keyring system asks for sender's key passphrase instead of receiver's. This I think is not really fixable.

**Comment by pravino** [comment]
@alecpl Yes iam on git master on Crypt_GPG 1.6.0b1
i agree with your flow ..
issue number 1 yes  this should be a bug
issue number two was just a test to see if it really wanted the senders private key.. on normal working scenarios no one will have the senders private key.

**Comment by alecpl** [comment]
Fixed in https://github.com/pear/Crypt_GPG/commit/67f6c5eade8025d8. I'll keep the ticket open until we have the next Crypt_GPG release.

**Comment by CRtEurope** [comment]
I reverted back to GPG 1.4.20 and the same issue. Also the key is not attached. The private key could not be found. : Decryption failed. Private key not found. Key ID: 6E629523.

All keys are available, public and private keys. It is also shown in settings page.

Using pear-pear.php.net/crypt_gpg (1.6.0b1)

```
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: USING GPG 1.4.20 with PHP 7.1.0RC6
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-use-agent --no-permission-warning --exit-on-status-write-error --trust-model always --homedir '/enigma/xxx@xxx.info' --decrypt
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: BEGIN PROCESSING
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG is ready for input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to write 6552 bytes to GPG input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => wrote 6552 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG is ready for input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to write 0 bytes to GPG input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => broken pipe on GPG input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => closing pipe GPG input
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 74 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: ENC_TO 775CA3FE6E629523 1 0
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: ENC_TO 5EECFD4F55A484F8 1 0
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 178 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: USERID_HINT 5EECFD4F55A484F8 Kai Kellner <kai.kellner@cesidian.info>
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: NEED_PASSPHRASE 5EECFD4F55A484F8 958825CD0A56A1BE 1 0
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GET_HIDDEN passphrase.enter
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG is ready for command data
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to write 1 bytes to GPG command
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => wrote 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 44 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GOT_IT
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: MISSING_PASSPHRASE
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 2
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG error stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG error
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 46 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: Invalid passphrase; please try again ...
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 219 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: BAD_PASSPHRASE 5EECFD4F55A484F8
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: USERID_HINT 5EECFD4F55A484F8 XXX XXX <xxx@xxx.info>
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: NEED_PASSPHRASE 5EECFD4F55A484F8 958825CD0A56A1BE 1 0
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GET_HIDDEN passphrase.enter
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG is ready for command data
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to write 1 bytes to GPG command
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => wrote 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 44 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GOT_IT
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: MISSING_PASSPHRASE
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 2
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG error stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG error
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 46 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: Invalid passphrase; please try again ...
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 219 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: BAD_PASSPHRASE 5EECFD4F55A484F8
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: USERID_HINT 5EECFD4F55A484F8 Kai Kellner <kai.kellner@cesidian.info>
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: NEED_PASSPHRASE 5EECFD4F55A484F8 958825CD0A56A1BE 1 0
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GET_HIDDEN passphrase.enter
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG is ready for command data
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to write 1 bytes to GPG command
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => wrote 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 1
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 44 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: GOT_IT
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: MISSING_PASSPHRASE
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 2
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG error stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG error
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 388 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: encrypted with 4096-bit RSA key, ID 55A484F8, created 2015-11-05
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR:       "XXX XXX <xxx@xxx.info>"
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: public key decryption failed: bad passphrase
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: encrypted with 4096-bit RSA key, ID 6E629523, created 2015-12-26
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR:       "XXXX) <xxx@xxx.eu>"
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: ERROR: gpg: decryption failed: secret key not available
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 189 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: BAD_PASSPHRASE 5EECFD4F55A484F8
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: ERROR pkdecrypt_failed 11
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: NO_SECKEY 775CA3FE6E629523
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: BEGIN_DECRYPTION
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: DECRYPTION_FAILED
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: STATUS: END_DECRYPTION
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: selecting streams
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => got 3
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG output stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG output
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 0 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG error stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG error
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 0 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: GPG status stream ready for reading
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => about to read 65536 bytes from GPG status
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => read 0 bytes
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: END PROCESSING
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: CLOSING GPG SUBPROCESS
[14-Nov-2016 14:19:42 +0100]: <s7sdsvf2> GPG: => subprocess returned an unexpected exit code: 2
```

**Comment by CRtEurope** [comment]
Used the patched version from  pear/Crypt_GPG@67f6c5e and now everything is working. 

Thanks very much for this patch @alecpl

**Comment by alecpl** [comment]
Crypt_GPG 1.6.0b2 released.
